docs/frontend-system/architecture/15-plugins.md

```diff
@@ -10,84 +10,81 @@ description: Frontend plugins
 
 ## Introduction
 
-<!--
+Frontend plugins are a foundational building block in Backstage and the frontend system. They are used to encapsulate and provide functionality for a Backstage app, such as new pages, navigational elements, APIs, as well as extensions and features for other plugins, such entity page cards and content for the Software Catalog, or result list items for the search plugin.
 
-Introduce frontend plugins and how each of them is shipped as a separate package. Highlight that each package should only export a single plugin instance.
-
--->
+Each plugin is typically shipped in a separate NPM package, whether that's a published package, or just in the local workspace. The plugins instance should always the `default` export of the package, either via the main entry-point or the `/alpha` sub-path export. Each plugin package is limited to exporting a single plugin instance. In a local workspace you could use a different structure if preferred, but this is considered a non-standard layout and should be avoided in published packages.
 
 ## Creating a Plugin
 
-<!--
-
-How to create a simple plugin
-
- -->
+Frontend plugin instances are created with the `createFrontendPlugin` function, which is provided by the `@backstage/frontend-plugin-api` package. It takes a single options object that provides all of the necessary configuration for the plugin. In particular you will want to provide [extensions](./20-extensions.md) for your plugin, as that is the way that you can provide new functionality to the app.
 
 ```ts
-export const myPlugin = createFrontendPlugin({
+// This creates a new extension, see "Extension Blueprints" documentation for more details
+const myPage = PageBlueprint.make({
+  params: {
+    defaultPath: '/my-page',
```

Comment:
Just noting that the loader is a required param. I know it's just an example, but you know these things can catch the reader off guard